examples/contrib/epidemiology/sir.py

```diff
@@ -11,26 +11,35 @@
 import torch
 
 import pyro
-from pyro.contrib.epidemiology import SimpleSIRModel
+from pyro.contrib.epidemiology import SimpleSEIRModel, SimpleSIRModel
 
 logging.basicConfig(format='%(message)s', level=logging.INFO)
 
 
+def Model(population, incubation_time, recovery_time, data):
+    """Dispatch between different model classes."""
+    if incubation_time:
+        assert incubation_time > 1
+        return SimpleSEIRModel(population, incubation_time, recovery_time, data)
+    return SimpleSIRModel(population, recovery_time, data)
+
+
 def generate_data(args):
     extended_data = [None] * (args.duration + args.forecast)
-    model = SimpleSIRModel(args.population, args.recovery_time, extended_data)
+    model = Model(args.population, args.incubation_time, args.recovery_time,
+                  extended_data)
     for attempt in range(100):
         samples = model.generate({"R0": args.basic_reproduction_number,
                                   "rho": args.response_rate})
         obs = samples["obs"][:args.duration]
-        S2I = samples["S2I"]
+        new_I = samples.get("S2I", samples.get("E2I"))
```

Comment:
what's going on here?

Reply:
In the SIR models, S2I is noisily observed. In SEIR models, E2I is noisily observed. This one-liner works for either model class.